osquery/dispatcher/scheduler.cpp

```diff
@@ -93,17 +92,23 @@ inline void launchQuery(const std::string& name, const ScheduledQuery& query) {
   // Comparisons and stores must include escaped data.
   sql.escapeResults();
 
+  Status status;
   DiffResults diff_results;
   // Add this execution's set of results to the database-tracked named query.
   // We can then ask for a differential from the last time this named query
   // was executed by exact matching each row.
-  auto status = dbQuery.addNewResults(sql.rows(), diff_results);
-  if (!status.ok()) {
-    std::string line = "Error adding new results to database: " + status.what();
-    LOG(ERROR) << line;
-
-    // If the database is not available then the daemon cannot continue.
-    Initializer::requestShutdown(EXIT_CATASTROPHIC, line);
+  if (!sql.eventBased()) {
```

Comment:
Fairly significant change here, the goal is to remove the call to `addNewResults` which stores the results in the DB. It also automatically computes the set difference.